pennylane/transforms/qmc.py

```diff
@@ -0,0 +1,353 @@
+    `this <https://arxiv.org/abs/1805.00109>`__ paper.
+
+    The multi-qubit gate :math:`Z = I - 2|0\rangle \langle 0|` can be performed using the
+    conventional multi-controlled-Z gate with an additional bit flip on each qubit before and after.
+
+    This function performs the multi-controlled-Z gate via a multi-controlled-X gate by picking an
+    arbitrary target wire to perform the X and adding a Hadamard on that wire either side of the
+    transformation.
+
+    Additional control from ``control_wire`` is then included within the multi-controlled-X gate.
+
+    Args:
+        wires (Wires): the wires on which the Z gate is applied
+        control_wire (Wires): the control wire from the register of phase estimation qubits
+        work_wires (Wires): the work wires used in the decomposition
+    """
+    target_wire = wires[0]
+    PauliX(target_wire)
+    Hadamard(target_wire)
+
+    control_values = "0" * (len(wires) - 1) + "1"
+    control_wires = wires[1:] + control_wire
+    MultiControlledX(
+        control_wires=control_wires,
+        wires=target_wire,
+        control_values=control_values,
+        work_wires=work_wires,
+    )
+
+    Hadamard(target_wire)
+    PauliX(target_wire)
+
+
+def _apply_controlled_v(target_wire, control_wire):
+    """Provides the circuit to apply a controlled version of the :math:`V` gate defined in
+    `this <https://arxiv.org/abs/1805.00109>`__ paper.
+
+    The :math:`V` gate is simply a Pauli-Z gate applied to the ``target_wire``, i.e., the ancilla
+    wire in which the expectation value is encoded.
+
+    The controlled version of this gate is then a CZ gate.
+
+    Args:
+        target_wire (Wires): the ancilla wire in which the expectation value is encoded
+        control_wire (Wires): the control wire from the register of phase estimation qubits
+    """
+    CZ(wires=[control_wire[0], target_wire[0]])
+
+
+def apply_controlled_Q(fn, wires, target_wire, control_wire, work_wires):
+    r"""Provides the circuit to apply a controlled version of the :math:`\mathcal{Q}` unitary
+    defined in `this <https://arxiv.org/abs/1805.00109>`__ paper.
+
+    Given a callable ``fn`` input corresponding to the :math:`\mathcal{F}` unitary in the above
+    paper, this function transforms the circuit into a controlled-version of the :math:`\mathcal{Q}`
+    unitary which forms part of the quantum Monte Carlo algorithm. In this algorithm, one of the
+    wires acted upon by :math:`\mathcal{F}`, specified by ``target_wire``, is used to embed a
+    Monte Carlo estimation problem. The :math:`\mathcal{Q}` unitary is then designed to encode the
+    target expectation value as a phase in one of its eigenvalues.
+
+    This function transforms to a controlled version of :math:`\mathcal{Q}` that is compatible with
+    quantum phase estimation (see :class:`~.QuantumPhaseEstimation` for more details).
+
+    Args:
+        fn (Callable): a quantum function that applies quantum operations according to the
```

Comment:
'quantum function' was originally defined on the glossary as a function that applies quantum operations, and returns a measurement statistic. I think it has since been removed, but mention of it remains in the [QNode entry](https://pennylane.ai/qml/glossary/quantum_node.html):

> A quantum node consists of a quantum function (such as a variational circuit), and a device on which it executes.

We've since started using the term a bit more loosely; the original definition of a quantum function was strictly

```python
def qfunc(params):
    quantum_operation()
    quantum_operation()
    quantum_operation()
    return measurement_statistic()
```

but we have since broadened it (in usage, not strictly in definition) to also include

```python
def qfunc(params):
    quantum_operation()
    quantum_operation()
    quantum_operation()
```